### PR TITLE
feat(cae/component): add resource to manage develop configurations

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Support two public (marshal and unmarshal) methods to build API requests for CAE service.
Support a new category and add a new convert method, then, unify all time-related methods into one file.
Add a new resource to manage develop configurations.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support two public methods to build API requests.
2. add a new category and add a new convert method.
3. unify all time-related methods into one file.
4. add resource to manage develop configurations.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
NONE
```

### CODE REVIEW

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/cae' TESTARGS='-run=TestAccComponentConfigurations_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cae -v -run=TestAccComponentConfigurations_basic -timeout 360m -parallel 4
=== RUN   TestAccComponentConfigurations_basic
=== PAUSE TestAccComponentConfigurations_basic
=== CONT  TestAccComponentConfigurations_basic
--- PASS: TestAccComponentConfigurations_basic (16.46s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cae       16.508s
```

File: docs/resources/cae_component_configurations.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    })
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the instance.
```
Comment (jinyangyang222) [inline docs/resources/cae_component_configurations.md:59]:
```suggestion
* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.  
```

File: huaweicloud/services/cae/resource_huaweicloud_cae_component_configurations.go [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+		MoreHeaders: map[string]string{
+			"X-Environment-ID": d.Get("environment_id").(string),
+		},
+	}
+
+	// Note: This API will delete all configurations under this component, regardless of whether it's configured
+	// through other channels.
+	_, err = client.Request("DELETE", deletePath, &deleteOpt)
+	if err != nil {
+		return diag.Errorf("error deleting configuration under the specific component (%s): %s", componentId, err)
```
Comment (jinyangyang222) [inline huaweicloud/services/cae/resource_huaweicloud_cae_component_configurations.go:310]:
```suggestion
		return diag.Errorf("error deleting configurations under the specific component (%s): %s", componentId, err)
```

File: docs/resources/cae_component_configurations.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the instance.  
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `environment_id` - (Required, String, ForceNew) Specifies the ID of the develop environment where the applications, applications
+  and components are located.
```
Comment (jinyangyang222) [inline docs/resources/cae_component_configurations.md:63]:
```suggestion
* `environment_id` - (Required, String, ForceNew) Specifies the ID of the develop environment where the applications and components are located.  
```

File: docs/resources/cae_component_configurations.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID (also component ID), in UUID format.
+
+## Import
+
+The resource can be imported using `environment_id`, `application_id` and `component_id`, e.g.
+
+```bash
+$ terraform import huaweicloud_cbh_instance.test <environment_id>/<application_id>/<component_id>
```
Comment (deer-hang) [inline docs/resources/cae_component_configurations.md]:
```suggestion
$ terraform import huaweicloud_cae_component_configurations.test <environment_id>/<application_id>/<component_id>
```

File: huaweicloud/services/cae/resource_huaweicloud_cae_component_configurations.go [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+		},
+	}
+	opts.JSONBody = utils.RemoveNil(buildCreateComponentConfigurationBodyParams(d))
+	// The creation API will return an 'operation_id' field, but this field is not helpful for resource logic, so it's
+	// ignored.
+	_, err = client.Request("POST", modifyPath, &opts)
+	if err != nil {
+		return diag.Errorf("error creating (overriding) the configuration for a component (%s): %s", componentId, err)
+	}
+	d.SetId(d.Get("component_id").(string))
```
Comment (deer-hang) [inline huaweicloud/services/cae/resource_huaweicloud_cae_component_configurations.go]:
```suggestion
	d.SetId(componentId)
```

File: docs/resources/cae_component_configurations.md (hunk 1 above)
Comment (Lance52259) [reply docs/resources/cae_component_configurations.md]: done.

File: huaweicloud/services/cae/resource_huaweicloud_cae_component_configurations.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/cae/resource_huaweicloud_cae_component_configurations.go]: done.

File: docs/resources/cae_component_configurations.md (hunk 3 above)
Comment (Lance52259) [reply docs/resources/cae_component_configurations.md]: done.

File: docs/resources/cae_component_configurations.md (hunk 4 above)
Comment (Lance52259) [reply docs/resources/cae_component_configurations.md]: done.

File: huaweicloud/services/cae/resource_huaweicloud_cae_component_configurations.go (hunk 5 above)
Comment (Lance52259) [reply huaweicloud/services/cae/resource_huaweicloud_cae_component_configurations.go]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/cae' TESTARGS='-run=TestAccComponentConfigurations_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cae -v -run=TestAccComponentConfigurations_basic -timeout 360m -parallel 4
=== RUN   TestAccComponentConfigurations_basic
=== PAUSE TestAccComponentConfigurations_basic
=== CONT  TestAccComponentConfigurations_basic
--- PASS: TestAccComponentConfigurations_basic (16.46s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cae       16.501s
```

**Pull request merged**